File: remove-element/remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        n = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[i], nums[n] = nums[n], nums[i]
                n += 1
        return n

File: remove-element/remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    var n = 0;
    var tmp;
    for(var i = 0; i < nums.length; i ++) {
        if (nums[i] != val) {
            tmp = nums[i];
            nums[i] = nums[n];
            nums[n] = tmp
            n += 1
        }
    }
    return n;
};
